apps/omg/test/omg/dependency_conformance/signature_test.exs

```diff
@@ -87,21 +103,13 @@ defmodule OMG.DependencyConformance.SignatureTest do
         metadata
       )
 
-    [alice_sig, bob_sig | _] = tx.sigs
-
-    verify(contract, tx, alice_sig)
-    verify(contract, tx, bob_sig)
+    verify(contract, tx)
   end
 
-  defp verify(contract, %Transaction.Signed{raw_tx: tx}, signature) do
-    {:ok, solidity_signer} =
-      Eth.call_contract(contract, "getSigner(bytes,bytes)", [Transaction.raw_txbytes(tx), signature], [:address])
-
-    {:ok, elixir_signer} =
-      tx
-      |> OMG.TypedDataHash.hash_struct()
-      |> OMG.Crypto.recover_address(signature)
+  defp verify(contract, %Transaction.Signed{raw_tx: tx}) do
+    {:ok, solidity_hash} =
+      Eth.call_contract(contract, "hashTx(address,bytes)", [contract, Transaction.raw_txbytes(tx)], [{:bytes, 32}])
 
-    assert solidity_signer == elixir_signer
+    assert solidity_hash == OMG.TypedDataHash.hash_struct(tx)
```

Comment:
are these results not predictable? why do we need an external source to verify the results in what should be a unit test?

Reply:
not really, given the pace at which the EIP712 changes :). Also the point of this test is to test against a particular "reference" impl, vide our discussion on Merkling elsewhere (#901).